yarn-project/aztec.js/src/account/entrypoint/eip_1271_account_entrypoint.ts

```diff
@@ -0,0 +1,104 @@
+    private partialAddress: PartialAddress,
+    private privateKey: PrivateKey,
+    private signer: Signer,
+    private chainId: number = DEFAULT_CHAIN_ID,
+    private version: number = DEFAULT_VERSION,
+  ) {}
+
+  /**
+   * Sign a message hash with the private key.
+   * @param message - The message hash to sign.
+   * @returns The signature as a Buffer.
+   */
+  public sign(message: Buffer): Buffer {
+    return this.signer.constructSignature(message, this.privateKey).toBuffer();
+  }
+
+  /**
+   * Creates an eip1271 witness for the given message. In this case, witness is the public key, the signature
+   * and the partial address, to be used for verification.
+   * @param message - The message hash to sign.
+   * @returns [publicKey, signature, partialAddress] as Fr[].
+   */
+  async createEip1271Witness(message: Buffer): Promise<Fr[]> {
+    const signature = this.sign(message);
+    const publicKey = await generatePublicKey(this.privateKey);
+
+    const sigFr: Fr[] = [];
+    for (let i = 0; i < 64; i++) {
+      sigFr.push(new Fr(signature[i]));
+    }
+
+    return [...publicKey.toFields(), ...sigFr, this.partialAddress];
+  }
+
+  /**
+   * Returns the transaction request and the eip1271 witness for the given function calls.
+   * Returning the witness here as a nonce is generated in the buildPayload action.
+   * @param executions - The function calls to execute
+   * @param opts - The options
+   * @returns The TxRequest, the eip1271 witness to insert in db and the message signed
+   */
+  async createTxExecutionRequestWithWitness(
+    executions: FunctionCall[],
+    opts: CreateTxRequestOpts = {},
+  ): Promise<{
+    /** The transaction request */
+    txRequest: TxExecutionRequest;
+    /** The eip1271 witness */
+    witness: Fr[];
+    /** The message signed */
+    message: Buffer;
+  }> {
+    if (opts.origin && !opts.origin.equals(this.address)) {
+      throw new Error(`Sender ${opts.origin.toString()} does not match account address ${this.address.toString()}`);
+    }
+
+    const { payload, packedArguments: callsPackedArguments } = await buildPayload(executions);
+    const message = await hashPayload(payload);
+    const witness = await this.createEip1271Witness(message);
+
+    const args = [payload];
+    const abi = this.getEntrypointAbi();
+    const packedArgs = await PackedArguments.fromArgs(encodeArguments(abi, args));
+    const txRequest = TxExecutionRequest.from({
+      argsHash: packedArgs.hash,
+      origin: this.address,
+      functionData: FunctionData.fromAbi(abi),
+      txContext: TxContext.empty(this.chainId, this.version),
+      packedArguments: [...callsPackedArguments, packedArgs],
+    });
+
+    return { txRequest, message, witness };
+  }
+
+  createTxExecutionRequest(executions: FunctionCall[], _opts: CreateTxRequestOpts = {}): Promise<TxExecutionRequest> {
+    throw new Error(`Not implemented, use createTxExecutionRequestWithWitness instead`);
+  }
```

Comment:
Not a fan of this since it breaks composability (though I'm fine with it until we further refine all this 1271 story). I'd extend the `TxExecutionRequest` to also contain a set of `MessageWithAuthWitness` (where a `MessageWithAuthWitness` is just key+value, ie message+witness), so that when it's run through the simulator, it picks witnesses from there and only uses the db as fallback. 

This should allow us to remove the new wallet type also.